### ISSUE TITLE
[tune] add **kwargs to Trial.__init__

### ISSUE THREAD

**Issue by joyyoj** [ray-project/ray#2711]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Ray installed from (source or binary)**:
- **Ray version**:
- **Python version**:
- **Exact command to reproduce**:

### Describe the problem

There is a function in config_parser.py:

    def create_trial_from_spec(spec, output_path, parser, **trial_kwargs):
        """Creates a Trial object from parsing the spec.

config_parser.create_trial_from_spec pass **trial_kwargs to Trial's constructor, but Trial.__init__ don't have argument **kwargs. So it fails if i pass some args those not in Trial.__init__. 
But some customized SearchAlgorithm may has  some extra information needed to be saved in Trial, so i think it makes sense to let Trial.__init__ accept **kwargs.

**Comment by richardliaw** [comment]
I think this isn't an issue right?

```
(ray) [rliaw@c73:/data/rliaw/explainability]$ cat test.py
from __future__ import print_function

def test(arg=None):
    return arg

values = {"arg": 1}
ret = test(**values)
print(ret)
(ray) [rliaw@c73:/data/rliaw/explainability]$ python test.py
1
(ray) [rliaw@c73:/data/rliaw/explainability]$ python2 test.py
1
```

**Comment by joyyoj** [comment]
Sorry that i didn't make myself clear.  I've updated the description of this issue.

**Comment by richardliaw** [comment]
Thanks!

**Comment by richardliaw** [comment]
Closed for now, as #2712 is closed.
